src/components/adopters/AdopterListWrapper.js

```diff
@@ -0,0 +1,116 @@
+import React, { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import SpinnerComponent from '../common/SpinnerComponent';
+import * as consts from '../../constants/apiConstants.js';
+import * as adopterActions from '../../actions/adopterActions';
+import AdopterListHeader from './AdopterListHeader';
+import AdopterList from './AdopterList';
+import '../../styles/animal-list.scss';
+
+class AdopterListWrapper extends Component {
+  constructor(props, context) {
+    super(props, context);
+
+    this.state = {
+      selectedAdopterId: '',
+      loading: true,
+      loadingList: true,
+      currPage: 1,
+      showBlacklist: false,
+      rows: consts.ADOPTER_PAGE_SIZE
+   };
+
+    this.onClick = this.onClick.bind(this);
+    this.onClickViewMore = this.onClickViewMore.bind(this);
+    this.onClickBlacklist = this.onClickBlacklist.bind(this);
+    this.onClickShowAll = this.onClickShowAll.bind(this);
+  }
+
+  componentWillMount() {
+    let { rows, currPage } = this.state;
+    this.props.actions.loadAdopters(rows, currPage);
+  }
+
+  componentWillReceiveProps(nextProps) {
+    this.setState({ loading: false, loadingList: false });
+    if (nextProps.adopters.firstPage) {
+      this.setState({ currPage: 1 });
+    }
+  }
+
+  componentWillUnmount() {
+    this.props.actions.cleanAdopters();
+  }
+
+  onClick(adopterId) {
+    const equalsId = this.state.selectedAdopterId === adopterId.toString();
+    this.setState({ selectedAdopterId: equalsId ? '' : adopterId.toString() });
+  }
+
+  onClickViewMore() {
+    let { rows, currPage, showBlacklist } = this.state;
+    let nextPage = currPage + 1;
+    this.setState({ currPage: nextPage });
+    this.setState({ loading: true });
+    if (showBlacklist) {
```

Comment:
seria lo mismo `this.props.actions.loadAdopters(rows, nextPage, showBlacklist);` que todo este if y else